manifests/init.pp

```diff
@@ -173,21 +173,21 @@
   Boolean $db_ssl_verify = true,
   String $db_name = 'candlepin',
   String $db_user = 'candlepin',
-  String $db_password = $candlepin::params::db_password,
+  Variant[Sensitive[String], String] $db_password = $candlepin::params::db_password,
   Stdlib::Absolutepath $crl_file = '/var/lib/candlepin/candlepin-crl.crl',
   Variant[Array[String], String] $user_groups = [],
   Stdlib::Absolutepath $log_dir = '/var/log/candlepin',
   String $oauth_key = 'candlepin',
   String $oauth_secret = 'candlepin',
   Boolean $env_filtering_enabled = true,
   Stdlib::Absolutepath $keystore_file = '/etc/candlepin/certs/keystore',
-  Optional[String] $keystore_password = undef,
+  Optional[Variant[Sensitive[String], String]] $keystore_password = undef,
   String $keystore_type = 'PKCS12',
   Stdlib::Absolutepath $truststore_file = '/etc/candlepin/certs/truststore',
-  Optional[String] $truststore_password = undef,
+  Optional[Variant[Sensitive[String], String]] $truststore_password = undef,
   Stdlib::Absolutepath $ca_key = '/etc/candlepin/certs/candlepin-ca.key',
   Stdlib::Absolutepath $ca_cert = '/etc/candlepin/certs/candlepin-ca.crt',
-  Optional[String] $ca_key_password = undef,
+  Optional[Variant[Sensitive[String], String]] $ca_key_password = undef,
```

Comment:
This looks like it needs to be unwrapped (or converted to EPP):
https://github.com/theforeman/puppet-candlepin/blob/db77ea2e365db7ddec2de268410fbe579dc61bff/templates/candlepin.conf.erb#L25-L26

Reply:
Yes, you a right.  I amended the necessary Change.